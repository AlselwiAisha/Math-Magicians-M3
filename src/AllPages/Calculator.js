import Calculator from '../Components/Calculator';

const Calc = () => (
  <Calculator />
);

export default Calc;
