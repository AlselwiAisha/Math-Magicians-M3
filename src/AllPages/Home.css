.main{
    width: 95%;
    height: 50vh;
    padding: 10px;
    display: flex;
    flex-direction: column;
    justify-content: space-around;
    text-align: left;
}

.main h1{
    margin: 2rem;
}