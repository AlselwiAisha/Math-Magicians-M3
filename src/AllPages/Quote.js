import Quote from '../Components/Quotes';

const Quotes = () => (
  <Quote />
);

export default Quotes;
