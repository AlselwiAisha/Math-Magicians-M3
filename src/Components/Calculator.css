.calcmain {
  display: flex;
  flex-direction: row;
  justify-content: space-around;
  align-items: center;
}

.calcmain h2 {
  font-size: 2.5rem;
  font-weight: 600;
  padding-bottom: 3rem;
}

.calcborder {
  margin: 2rem;
}

.clcBckgrnd {
  background-color: #f4f3f3;
  border-radius: 0.3rem;
  margin-top: 4rem;
  padding: 1rem;
  width: 30rem;
  box-shadow: 0 9px 15px -3px rgba(0, 0, 0, 0.1);
}

.clcScrn {
  font-size: 2rem;
  font-weight: 600;
  text-align: right;
  background-color: #858694;
  color: #e9f0f4;
  height: 4rem;
  margin-bottom: 0.1rem;
  padding: 0.8rem;
}

.btnsPad {
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  gap: 0.1rem;
}

.btn {
  display: flex;
  align-items: center;
  justify-content: center;
  background-color: #e0e0e0;
  color: #242424;
  font-size: 1.1rem;
  font-weight: 500;
  border: none;
  height: 3.5rem;
  padding: 10px;
  cursor: pointer;
}

.btn:hover {
  border: 2px solid #65737e;
}

.zero {
  grid-column: 1/3;
}

.btn2 {
  background: #f5913e;
}
