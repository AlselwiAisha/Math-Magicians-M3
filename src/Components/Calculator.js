import React, { useState } from 'react';
import './Calculator.css';
import ClcScrn from './ClcScrn';
import Btn from './Btn';
import calculate from '../logic/calculate';

const Calculator = () => {
  const [total, setTotal] = useState(0);
  const [next, setNext] = useState('');
  const [operation, setOperation] = useState('');

  const onClick = (e) => {
    const btn = e.target.innerHTML;
    const results = calculate({ total, next, operation }, btn);
    setTotal(results.total || '');
    setNext(results.next || '');
    setOperation(results.operation || '');
  };
  const btnValues = [
    ['AC', '+/-', '%', 'รท'],
    ['7', '8', '9', 'x'],
    ['4', '5', '6', '-'],
    ['1', '2', '3', '+'],
    ['0', '.', '='],
  ];
  const dis = "Let's do some math!";
  return (
    <div className="calcmain">
      <h2>{dis}</h2>
      <div className="calcborder">
        <div className="clcBckgrnd">
          <ClcScrn total={total} next={next} operation={operation} />
          <div key={(i) => i + 1} className="btnsPad">
            {btnValues.flat().map((row) => (
              <Btn
                key={row}
                value={row}
                onClick={onClick}
              />
            ))}
          </div>
        </div>
      </div>
    </div>
  );
};
export default Calculator;
