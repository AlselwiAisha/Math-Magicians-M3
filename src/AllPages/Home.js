import './Home.css';

const Home = () => (
  <div className="main">
    <h1>Welcom to our page</h1>
    <p>
      Lorem ipsum dolor, sit amet consectetur adipisicing elit. Repellat,
      alias. Recusandae labore architecto aspernatur ipsa asperiores,
      placeat aperiam odio praesentium, exercitationem, temporibus et
      dolorem repellat molestias sint reiciendis voluptatibus. Autem?
      Lorem ipsum dolor, sit amet consectetur adipisicing elit. Repellat,
      alias. Recusandae labore architecto aspernatur ipsa asperiores,
      placeat aperiam odio praesentium, exercitationem, temporibus et dolorem
      repellat molestias sint reiciendis voluptatibus. Autem?
    </p>
    <p>
      Lorem ipsum dolor, sit amet consectetur adipisicing elit. Repellat, alias.
      Recusandae labore architecto aspernatur ipsa asperiores,
      placeat aperiam odio praesentium, exercitationem, temporibus
      et dolorem repellat molestias sint reiciendis voluptatibus. Autem?
      Lorem ipsum dolor, sit amet consectetur adipisicing elit. Repellat,
      alias. Recusandae labore architecto aspernatur ipsa asperiores,
      placeat aperiam odio praesentium, exercitationem, temporibus et
      dolorem repellat molestias sint reiciendis voluptatibus. Autem?
    </p>
  </div>
);
export default Home;
