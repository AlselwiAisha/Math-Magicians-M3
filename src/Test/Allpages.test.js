import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Home from '../AllPages/Home';
import Calculator from '../AllPages/Calculator';
import Quote from '../AllPages/Quote';

describe('Home AllPages', () => {
  it('should match the snapshot', () => {
    const { container } = render(<Home />);
    expect(container).toMatchSnapshot();
  });
});
describe('Calculator AllPages', () => {
  it('should match the snapshot', () => {
    const { container } = render(<Calculator />);
    expect(container).toMatchSnapshot();
  });
});
describe('Quote AllPages', () => {
  it('should match the snapshot', () => {
    const { container } = render(<Quote />);
    expect(container).toMatchSnapshot();
  });
});
