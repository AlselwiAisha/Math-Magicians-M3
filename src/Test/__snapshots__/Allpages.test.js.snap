// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator AllPages should match the snapshot 1`] = `
<div>
  <div
    class="calcmain"
  >
    <h2>
      Let's do some math!
    </h2>
    <div
      class="calcborder"
    >
      <div
        class="clcBckgrnd"
      >
        <div
          class="clcScrn"
        >
          0
        </div>
        <div
          class="btnsPad"
        >
          <button
            class="num btn"
            type="button"
          >
            AC
          </button>
          <button
            class="num btn"
            type="button"
          >
            +/-
          </button>
          <button
            class="num btn"
            type="button"
          >
            %
          </button>
          <button
            class="btn2 btn"
            type="button"
          >
            รท
          </button>
          <button
            class="num btn"
            type="button"
          >
            7
          </button>
          <button
            class="num btn"
            type="button"
          >
            8
          </button>
          <button
            class="num btn"
            type="button"
          >
            9
          </button>
          <button
            class="btn2 btn"
            type="button"
          >
            x
          </button>
          <button
            class="num btn"
            type="button"
          >
            4
          </button>
          <button
            class="num btn"
            type="button"
          >
            5
          </button>
          <button
            class="num btn"
            type="button"
          >
            6
          </button>
          <button
            class="btn2 btn"
            type="button"
          >
            -
          </button>
          <button
            class="num btn"
            type="button"
          >
            1
          </button>
          <button
            class="num btn"
            type="button"
          >
            2
          </button>
          <button
            class="num btn"
            type="button"
          >
            3
          </button>
          <button
            class="btn2 btn"
            type="button"
          >
            +
          </button>
          <button
            class="zero btn"
            type="button"
          >
            0
          </button>
          <button
            class="num btn"
            type="button"
          >
            .
          </button>
          <button
            class="btn2 btn"
            type="button"
          >
            =
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Home AllPages should match the snapshot 1`] = `
<div>
  <div
    class="main"
  >
    <h1>
      Welcom to our page
    </h1>
    <p>
      Lorem ipsum dolor, sit amet consectetur adipisicing elit. Repellat, alias. Recusandae labore architecto aspernatur ipsa asperiores, placeat aperiam odio praesentium, exercitationem, temporibus et dolorem repellat molestias sint reiciendis voluptatibus. Autem? Lorem ipsum dolor, sit amet consectetur adipisicing elit. Repellat, alias. Recusandae labore architecto aspernatur ipsa asperiores, placeat aperiam odio praesentium, exercitationem, temporibus et dolorem repellat molestias sint reiciendis voluptatibus. Autem?
    </p>
    <p>
      Lorem ipsum dolor, sit amet consectetur adipisicing elit. Repellat, alias. Recusandae labore architecto aspernatur ipsa asperiores, placeat aperiam odio praesentium, exercitationem, temporibus et dolorem repellat molestias sint reiciendis voluptatibus. Autem? Lorem ipsum dolor, sit amet consectetur adipisicing elit. Repellat, alias. Recusandae labore architecto aspernatur ipsa asperiores, placeat aperiam odio praesentium, exercitationem, temporibus et dolorem repellat molestias sint reiciendis voluptatibus. Autem?
    </p>
  </div>
</div>
`;

exports[`Quote AllPages should match the snapshot 1`] = `
<div>
  <div
    class="Container"
  >
    <p
      class="loading"
    >
      Loading ...
    </p>
  </div>
</div>
`;
